float64[] del_s
float64[] fx_m